### PR TITLE
fix(markdownlint): flag relative paths in Markdown links

### PR DESCRIPTION
**Pull request by OnkarRuikar**
### Description

Addresses: https://github.com/mdn/content/pull/21030#issuecomment-1257379857

### Motivation

Contributors try to use relative paths in the Markdown style links, considering the `index.md` file's location:
`[well documented](../../issues/content_suggestions_feature_proposals)`

But the link fails cos yari expects paths to be relative as per the parent directory:
`[well documented](../issues/content_suggestions_feature_proposals)`

Apart from this, the relative links easily break when the document is moved, renamed, or parent is renamed etc.
It's better to avoid use of relative paths in the links.

### Solution
We've [converted all the relative links](https://github.com/mdn/content/pull/21030) (`../../`), in the content. Now we can simply add this custom rule to flag any future uses.

As the linter can not convert relative path to absolute path, as per yari's requirements, flagging such uses is enough. So that contributors will fix them manually.

cc/ @teoli2003 , @nschonni

### CODE REVIEW

File: .markdownlint-cli2.jsonc [hunk 1]
```diff
@@ -152,6 +152,12 @@
           "replace": "(/en-US/docs",
           "skipCode": true
         },
+        {
+          "name": "relative-link-path",
+          "message": "Don't use relative paths",
+          "search": "](..",
```
Comment (nschonni) [inline .markdownlint-cli2.jsonc:158]: Do these need to be escaped to be valid regex `\]\(\.\.` or does it also do regular string matching

Comment (OnkarRuikar) [reply .markdownlint-cli2.jsonc:158]: The option `search` does text match and `searchPattern` does regex match. [[ref](https://github.com/OnkarRuikar/markdownlint-rule-search-replace#using-markdownlintjson-config-file)]

Comment (Josh-Cena) [inline .markdownlint-cli2.jsonc:158]: Can we merge this rule with the previous rule to detect all links like `\]\((?!https?|/en-US/)`? Or does it not support regexes?

**Comment by teoli2003** [comment]
For the moment, we only have automtically fixable rules. If we add rules that must be manually fixed, we should discuss how to do it, as this changes the flow. For the moment, the result of Markdownlint in CI can be ignored as there is a daily markdownlint fix. This won't be the case anymore (or with the rule we want to add to detect missing alt). Maybe we should have several configs? One used by the daily Markdownlint PR (With only auto-fixable problems); one by editors like VS Code (with everything), and one with only non-fixable problems used in the CI ? Or, maybe this is fine like this... I think an MDN discussion about this would be useful.

File: .markdownlint-cli2.jsonc (hunk 1 above)
Comment (OnkarRuikar) [reply .markdownlint-cli2.jsonc:158]: Previous rule has a replacement, but this one doesn't. Replacement helps doing auto fix `yarn fix:md`.
We can do this only if we can find a common replacement pattern.

**Comment by nschonni** [comment]
The job already handles the case where if hits rules it can't fix, as not all Markdown rules can be auto-fixed
https://github.com/mdn/content/blob/3c67eed7b0d2c91198ec95bcc9b1a570f6c0f585/.github/workflows/markdown-lint-fix.yml#L40-L50

**Review by caugner — APPROVED** [review]
LGTM once the test change has been reverted.

**Review by caugner — APPROVED** [review]

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**